Directory.Build.props

```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <LangVersion>7.2</LangVersion>
     <NoWarn>$(NoWarn);1573;1591</NoWarn>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```

Comment:
💡 (but this is a strong recommendation so closer to ❗️ ) Consider changing this to only treat warnings as errors for command line and AppVeyor builds. This will avoid interfering with `F5` to debug and the ability to run unit tests while working inside Visual Studio.

```xml
  <PropertyGroup Condition="'$(BuildingInsideVisualStudio)' != 'true'">
    <!-- Ideally this is always enabled, but that tends to hurt developer productivity -->
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>
```

Reply:
But by doing so, developers won't know that build has failed until after they have committed the code...

Reply:
@sharwell am I right in assuming you mean the developer would need to verify the build output didn't contain warnings before submitting a PR?

For myself, I would rather have the computer put a hard block in my way so that I stay on top of warnings/errors as they come up. It also removes the chance of sending off a PR and having it fail CI, which saves me some time (and face 😀).

Reply:
OK to keep it on.
I will be _very_ annoyed by this, but at least I can turn it of locally.
I still have the problem in #4467 so I always have warnings.